cdap-ui/app/cdap/components/Overview/AppOverview/index.js

```diff
@@ -18,25 +18,64 @@ import React, {PropTypes, Component} from 'react';
 import isNil from 'lodash/isNil';
 import OverviewHeader from 'components/Overview/OverviewHeader';
 import OverviewMetaSection from 'components/Overview/OverviewMetaSection';
+import AppOverviewTab from 'components/Overview/AppOverview/AppOverviewTab';
+import {MyAppApi} from 'api/app';
+import NamespaceStore from 'services/NamespaceStore';
+import {objectQuery} from 'services/helpers';
 
 require('./AppOverview.scss');
 
 export default class AppOverview extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      entity: this.props.entity
+      entity: this.props.entity,
+      activeTab: '1',
+      entityDetail: null,
+      loading: false
     };
   }
+  componentWillMount() {
+    this.fetchAppDetail();
+  }
   componentWillReceiveProps(nextProps) {
     let {entity} = nextProps;
     if (!isNil(entity)) {
       this.setState({
-        entity
-      });
+        entity,
+      }, this.fetchAppDetail.bind(this));
+    }
+  }
+  fetchAppDetail() {
+    this.setState({
+      loading: true
+    });
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    if (objectQuery(this.props, 'entity', 'id')) {
+      MyAppApi
+        .get({
+          namespace,
+          appId: this.props.entity.id
+        })
+        .subscribe(res => {
+          this.setState({
+            entityDetail: res
+          }, () => {
+            setTimeout(() => {
```

Comment:
This is to smooth out hiding loading icon. If the backend sends a response immediately this is to make sure the transition is smooth rather than snappy. 

The reason this is the second argument for setState is I want to set the entityDetail for rendering and wait for a second before hiding loading icon.  Could just move it out too except didn't want to have both two `setState`s as it will be batched together by react.